src/hotspot/share/opto/c2_CodeStubs.hpp

```diff
@@ -95,6 +95,19 @@ class C2CheckLockStackStub : public C2CodeStub {
 };
 
 #ifdef _LP64
+class C2FixAnonOMOwnerStub : public C2CodeStub {
```

Comment:
I dislike the name a bit since it does more than fix the owner in monitor, it also pops one slot from the lock stack.